src/index/id_map.rs

```diff
@@ -364,6 +366,21 @@ where
     }
 }
 
+impl IndexImpl {
+    pub fn into_id_map(self) -> Result<IdMap<IndexImpl>> {
```

Comment:
A bit of documentation and it's good to go.
```suggestion
    /// Attempt a dynamic cast of the index to one that is [ID-mapped][1].
    /// 
    /// [1]: crate::IdMap
    pub fn into_id_map(self) -> Result<IdMap<IndexImpl>> {
```